core/spike/common/syscall.c

```diff
@@ -0,0 +1,507 @@
+// See LICENSE for license details.
+
+#include <stdint.h>
+#include <string.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <limits.h>
+#include <sys/signal.h>
+#include "util.h"
+
+#define SYS_write 64
+
+#undef strcmp
+
+extern volatile uint64_t tohost;
+extern volatile uint64_t fromhost;
+
+static uintptr_t syscall(uintptr_t which, uint64_t arg0, uint64_t arg1, uint64_t arg2)
+{
```

Comment:
This code formatting style is different from the existing code in reactor-c. We put `{`s on the same line.

Also, comments are missing for some of the files and the functions within those files.

Reply:
All of the spike/common files came from https://github.com/riscv-software-src/riscv-tests/tree/master/benchmarks/common  I don't want to mess with them too much, so I've just added a link to the original source + licence at the top of each file.